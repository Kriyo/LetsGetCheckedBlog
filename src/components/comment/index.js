import React from 'react'

import {
  Container,
  Item,
  RootItem,
  NestedItem,
  Count,
  Image,
  DetailsWrapper,
  Content,
  AuthorDetails,
  ReplyHeading,
  Form,
  Button,
  ButtonWrapper,
  ErrorLabel,
  Input,
  TextArea,
  Section,
} from './styles/comment'

export const Comment = ({ children, ...props }) => {
  return <Item {...props}>{children}</Item>
}

Comment.Frame = ({ children, ...props }) => {
  return <Container {...props}>{children}</Container>
}

Comment.Section = ({ children, ...props }) => {
  return <Section {...props}>{children}</Section>
}

Comment.RootComment = ({ children, ...props }) => {
  return <RootItem {...props}>{children}</RootItem>
}

Comment.NestedComment = ({ children, ...props }) => {
  return <NestedItem {...props}>{children}</NestedItem>
}

Comment.Count = ({ children, ...props }) => {
  return <Count {...props}>{children}</Count>
}

Comment.DetailsWrapper = ({ children, ...props }) => {
  return <DetailsWrapper {...props}>{children}</DetailsWrapper>
}

Comment.AuthorDetails = ({ children, ...props }) => {
  return <AuthorDetails {...props}>{children}</AuthorDetails>
}

Comment.Avatar = ({ children, ...props }) => {
  return <Image {...props}>{children}</Image>
}

Comment.Content = ({ children, ...props }) => {
  return <Content {...props}>{children}</Content>
}

Comment.ReplyHeading = ({ children, ...props }) => {
  return <ReplyHeading {...props}>{children}</ReplyHeading>
}

Comment.Form = ({ children, ...props }) => {
  return <Form {...props}>{children}</Form>
}

Comment.ButtonWrapper = ({ children, ...props }) => {
  return <ButtonWrapper {...props}>{children}</ButtonWrapper>
}

Comment.Button = ({ children, ...props }) => {
  return <Button {...props}>{children}</Button>
}
Comment.ErrorLabel = ({ children, ...props }) => {
  return <ErrorLabel {...props}>{children}</ErrorLabel>
}
Comment.UserInput = ({ children, ...props }) => {
  return <Input {...props}>{children}</Input>
}

Comment.TextArea = ({ children, ...props }) => {
  return <TextArea {...props}>{children}</TextArea>
}
