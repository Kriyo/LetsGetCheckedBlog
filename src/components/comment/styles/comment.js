import styled from 'styled-components/macro'
// import { Primary } from '../../../utils/colors'

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  max-width: 600px;
  width: 100%;
  margin: 0 auto;
`

export const Item = styled.div``

export const RootItem = styled.div`
  display: flex;
  max-width: 540px;
  justify-content: space-between;
  @media (max-width: 600px) {
    flex-direction: row;
    width: 90%;
    margin: 0 auto;
  }
`

export const NestedItem = styled.div`
  display: flex;
  max-width: 540px;
  justify-content: space-between;
  padding: 10px 0;
  margin-left: 50px;
  @media (max-width: 600px) {
    flex-direction: row;
    width: 90%;
    margin: 0 auto;
  }
`

export const Count = styled.h2`
  @media (max-width: 600px) {
    margin: 20px;
  }
`

export const Section = styled.div``

export const DetailsWrapper = styled.div`
  display: flex;
  flex-direction: column;
  padding: 0 20px;
`

export const AuthorDetails = styled.h4`
  margin: 0;
`

export const Image = styled.img`
  width: 50px;
  height: 50px;
`

export const Content = styled.p``

export const ReplyHeading = styled.h3``

export const Form = styled.form`
  display: flex;
  flex-direction: column;

  @media (max-width: 600px) {
    padding: 15px;
  }
`

export const ButtonWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 0 auto;
`

export const Button = styled.button`
  height: 35px;
  width: 80px;
`

export const Input = styled.input`
  padding: 8px 12px;
  width: 185px;
  margin: 10px 0;
  background: #fff;
  border: 1px solid #d9d9d9;
  border-top: 1px solid #c0c0c0;
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  -moz-border-radius: 1px;
  -webkit-border-radius: 1px;
  border-radius: 1px;
  font-size: 0.9rem;
  color: #404040;
  outline: none;

  &[type='submit'] {
    cursor: pointer;
  }
  @media (max-width: 600px) {
    width: 100%;
  }
`

export const TextArea = styled.textarea`
  font-family: 'Open Sans', sans-serif;
  padding: 8px 12px;
  height: 150px;
  width: 500px;
  margin: 10px 0;
  background: #fff;
  border: 1px solid #d9d9d9;
  border-top: 1px solid #c0c0c0;
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  -moz-border-radius: 1px;
  -webkit-border-radius: 1px;
  border-radius: 1px;
  font-size: 0.9rem;
  color: #404040;
  outline: none;

  @media (max-width: 600px) {
    width: 100%;
    margin: auto;
  }
`
