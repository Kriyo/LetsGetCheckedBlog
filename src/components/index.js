export { Comment } from './comment'
export { Feature } from './feature'
export { NavBar } from './navbar'
export { Post } from './post'
