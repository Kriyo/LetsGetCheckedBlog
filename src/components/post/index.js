import React from 'react'

import {
  Container,
  Item,
  Author,
  Title,
  Description,
  PublishDate,
  Content,
} from './styles/post'

export const Post = ({ children, ...props }) => {
  return <Item {...props}>{children}</Item>
}

Post.Frame = ({ children, ...props }) => {
  return <Container {...props}>{children}</Container>
}

Post.Author = ({ children, ...props }) => {
  return <Author {...props}>{children}</Author>
}

Post.Title = ({ children, ...props }) => {
  return <Title {...props}>{children}</Title>
}

Post.Description = ({ children, ...props }) => {
  return <Description {...props}>{children}</Description>
}

Post.PublishDate = ({ children, ...props }) => {
  return <PublishDate {...props}>{children}</PublishDate>
}

Post.Content = ({ children, ...props }) => {
  return <Content {...props}>{children}</Content>
}
