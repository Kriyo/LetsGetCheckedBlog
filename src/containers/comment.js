import React from 'react'
import { Comment } from '../components'

export const CommentContainer = () => {
  return (
    <Comment.Frame>
      <Comment.Count>2</Comment.Count>
    </Comment.Frame>
  )
}
