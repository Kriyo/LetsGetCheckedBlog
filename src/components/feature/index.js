import React from 'react'

import { Container, Title, SubTitle } from './styles/feature'

export const Feature = ({ children, ...props }) => {
  return <Container {...props}>{children}</Container>
}

Feature.Title = ({ children, ...props }) => {
  return <Title {...props}>{children}</Title>
}

Feature.SubTitle = ({ children, ...props }) => {
  return <SubTitle {...props}>{children}</SubTitle>
}
