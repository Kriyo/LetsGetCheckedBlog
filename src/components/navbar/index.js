import React from 'react'
import { Background, Container, Link, Logo } from './styles/navbar'

export default function NavBar({ background = true, children, ...props }) {
  return background ? <Background {...props}>{children}</Background> : children
}

NavBar.Frame = function NavBarFrame({ children, ...props }) {
  return <Container {...props}>{children}</Container>
}

NavBar.Link = function NavBarLink({ children, ...props }) {
  return <Link {...props}>{children}</Link>
}

NavBar.Logo = function NavBarLogo({ to, ...restProps }) {
  return <Logo {...restProps} />
}
