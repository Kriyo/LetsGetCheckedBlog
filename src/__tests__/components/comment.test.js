import React from 'react'
import { render } from '@testing-library/react'
import { Comment } from '../../components/comment'
import avatar from '../../default-avatar.png'

describe('<Comment />', () => {
  it('renders the <Comment /> with populated data', () => {
    const { container, getByText } = render(
      <Comment.Frame>
        <Comment.Count>5 Comments</Comment.Count>
        <Comment.Section>
          <Comment>
            <Comment.RootComment>
              <Comment.Avatar src={avatar} />
              <Comment.DetailsWrapper>
                <Comment.AuthorDetails>
                  Chris on 2021-03-03
                </Comment.AuthorDetails>
                <Comment.Content>
                  Ius ad verear tincidunt scriptorem
                </Comment.Content>
              </Comment.DetailsWrapper>
              <Comment.ButtonWrapper>
                <Comment.Button onClick={() => {}}>Reply</Comment.Button>
              </Comment.ButtonWrapper>
            </Comment.RootComment>
            <Comment.NestedComment>
              <Comment.Avatar src={avatar} />
              <Comment.DetailsWrapper>
                <Comment.AuthorDetails>
                  Jason response to Chris on 2021-03-04
                </Comment.AuthorDetails>
                <Comment.Content>Purto tamquam labores has ex.</Comment.Content>
              </Comment.DetailsWrapper>
              <Comment.ButtonWrapper>
                <Comment.Button onClick={() => {}}>Reply</Comment.Button>
              </Comment.ButtonWrapper>
            </Comment.NestedComment>
          </Comment>
        </Comment.Section>
        <Comment.Form onSubmit={() => {}}>
          <Comment.ReplyHeading>Submit A Comment</Comment.ReplyHeading>
          <Comment.Content>
            Reply to Chris
            <Comment.Button onClick={() => {}}>Cancel</Comment.Button>
          </Comment.Content>
          <Comment.UserInput
            type="text"
            id="username"
            placeholder="Your Name"
            value=""
            onChange={() => {}}
          />

          <Comment.ErrorLabel htmlFor="username">
            Username is too short
          </Comment.ErrorLabel>
          <Comment.TextArea
            id="usercomment"
            placeholder="Your Comment"
            value=""
            onChange={() => {}}
          />
          <Comment.ErrorLabel htmlFor="usercomment">
            Comment is too short
          </Comment.ErrorLabel>
          <Comment.UserInput disabled type="submit" value="Submit" />
        </Comment.Form>
      </Comment.Frame>
    )

    expect(getByText('5 Comments')).toBeTruthy()
    expect(getByText('Chris on 2021-03-03')).toBeTruthy()
    expect(getByText('Ius ad verear tincidunt scriptorem')).toBeTruthy()
    expect(getByText('Jason response to Chris on 2021-03-04')).toBeTruthy()
    expect(getByText('Purto tamquam labores has ex.')).toBeTruthy()
    expect(getByText('Submit A Comment')).toBeTruthy()
    expect(getByText('Reply to Chris')).toBeTruthy()
    expect(getByText('Cancel')).toBeTruthy()
    expect(getByText('Username is too short')).toBeTruthy()
    expect(getByText('Comment is too short')).toBeTruthy()
    expect(getByText('Submit')).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })
})
