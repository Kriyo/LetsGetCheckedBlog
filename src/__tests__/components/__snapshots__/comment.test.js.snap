// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comment /> renders the <Comment /> with populated data 1`] = `
<div
  class="comment__Container-sc-1vl0att-0 eSWrdR"
>
  <h2
    class="comment__Count-sc-1vl0att-4 dBntRH"
  >
    5 Comments
  </h2>
  <div
    class="comment__Section-sc-1vl0att-5"
  >
    <div
      class="comment__Item-sc-1vl0att-1"
    >
      <div
        class="comment__RootItem-sc-1vl0att-2 efLBJg"
      >
        <img
          class="comment__Image-sc-1vl0att-8 bpzdjn"
          src="default-avatar.png"
        />
        <div
          class="comment__DetailsWrapper-sc-1vl0att-6 kprSpY"
        >
          <h4
            class="comment__AuthorDetails-sc-1vl0att-7 FPVPu"
          >
            Chris on 2021-03-03
          </h4>
          <p
            class="comment__Content-sc-1vl0att-9 dYpyMh"
          >
            Ius ad verear tincidunt scriptorem
          </p>
        </div>
        <div
          class="comment__ButtonWrapper-sc-1vl0att-12 iuyUmH"
        >
          <button
            class="comment__Button-sc-1vl0att-13 bYslNS"
          >
            Reply
          </button>
        </div>
      </div>
      <div
        class="comment__NestedItem-sc-1vl0att-3 dgizcD"
      >
        <img
          class="comment__Image-sc-1vl0att-8 bpzdjn"
          src="default-avatar.png"
        />
        <div
          class="comment__DetailsWrapper-sc-1vl0att-6 kprSpY"
        >
          <h4
            class="comment__AuthorDetails-sc-1vl0att-7 FPVPu"
          >
            Jason response to Chris on 2021-03-04
          </h4>
          <p
            class="comment__Content-sc-1vl0att-9 dYpyMh"
          >
            Purto tamquam labores has ex.
          </p>
        </div>
        <div
          class="comment__ButtonWrapper-sc-1vl0att-12 iuyUmH"
        >
          <button
            class="comment__Button-sc-1vl0att-13 bYslNS"
          >
            Reply
          </button>
        </div>
      </div>
    </div>
  </div>
  <form
    class="comment__Form-sc-1vl0att-11 fBrpNf"
  >
    <h3
      class="comment__ReplyHeading-sc-1vl0att-10"
    >
      Submit A Comment
    </h3>
    <p
      class="comment__Content-sc-1vl0att-9 dYpyMh"
    >
      Reply to Chris
      <button
        class="comment__Button-sc-1vl0att-13 bYslNS"
      >
        Cancel
      </button>
    </p>
    <input
      class="comment__Input-sc-1vl0att-15 dbQFyA"
      id="username"
      placeholder="Your Name"
      type="text"
      value=""
    />
    <label
      class="comment__ErrorLabel-sc-1vl0att-14 Ldqhp"
      for="username"
    >
      Username is too short
    </label>
    <textarea
      class="comment__TextArea-sc-1vl0att-16 czCmQW"
      id="usercomment"
      placeholder="Your Comment"
    />
    <label
      class="comment__ErrorLabel-sc-1vl0att-14 Ldqhp"
      for="usercomment"
    >
      Comment is too short
    </label>
    <input
      class="comment__Input-sc-1vl0att-15 dbQFyA"
      disabled=""
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;
